Add test coverage of deserialisation of skin layouts

peppy opened the pull request:
Follow-up to https://github.com/ppy/osu/pull/19481.

peppy committed:
Add test coverage of deserialisation skin layouts

peppy committed:
Add test skin layout json resources

frenzibyte commented on osu.Game.Tests/Skins/SkinDeserialisationTest.cs:
```diff
@@ -0,0 +1,78 @@
+    /// If this test breaks, check any naming or class structure changes.
+    /// Migration rules may need to be added to <see cref="Skin"/>.
+    /// </remarks>
+    [TestFixture]
+    public class SkinDeserialisationTest
```
May be worthwhile to have this test also fail when new skinnable drawables are added to the game (likely using reflection), to cover against them when they're later on renamed or otherwise.

peppy replied:
I've added quite a bit more coverage.

frenzibyte reviewed (commented):
How would one go on updating these skins? i.e. is there a special reason they're suffixed with a date?

peppy committed:
Apply nullability to `SkinComponentToolbox` and split out reflection method to get all skinnable components

peppy committed:
Improve test coverage of skin serialisation to ensure full coverage

Will fail when new skinnable components are added until they have
coverage in resources.

peppy committed:
Add test coverage of remaining components which weren't already included

peppy commented:
> i.e. is there a special reason they're suffixed with a date?

We version most things by date so it felt correct.

smoogipoo committed:
Remove unused using
